src/components/[guild]/RolePlatforms/components/AddRoleRewardModal/components/DynamicSetup/BaseValueModal.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  ModalBody,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Spinner,
+  Stack,
+  Text,
+  VStack,
+} from "@chakra-ui/react"
+import LogicDivider from "components/[guild]/LogicDivider"
+import RequirementDisplayComponent from "components/[guild]/Requirements/components/RequirementDisplayComponent"
+import useRequirements from "components/[guild]/hooks/useRequirements"
+import DisplayCard from "components/common/DisplayCard"
+import { Modal } from "components/common/Modal"
+import AddRequirement from "components/create-guild/Requirements/components/AddRequirement"
+import useCreateRequirement from "components/create-guild/Requirements/hooks/useCreateRequirement"
+import { CaretRight } from "phosphor-react"
+import { useFormContext } from "react-hook-form"
+import { REQUIREMENT_PROVIDED_VALUES } from "requirements/requirements"
+import { Requirement } from "types"
+
+type Props = {
+  roleId: number
+  isOpen: boolean
+  onClose: () => void
+  onSelect: (reqId: number) => void
+}
+
+const BaseValueModal = ({ roleId, isOpen, onClose, onSelect }: Props) => {
+  const { data: requirements, mutate, isLoading } = useRequirements(roleId)
+
+  const dynamicRequirements =
+    requirements?.filter((req) => !!REQUIREMENT_PROVIDED_VALUES[req.type]) || []
+
+  const {
+    onSubmit: onCreateRequirementSubmit,
+    isLoading: isCreateRequirementLoading,
+  } = useCreateRequirement(roleId, {
+    onSuccess: () => {
+      mutate()
+    },
+    onError: () => {},
+  })
+
+  const addRequirement = (req: Requirement) => {
+    onCreateRequirementSubmit(req)
+  }
+
+  const Loader = () => (
+    <VStack my={5}>
+      <Spinner />
+      <Text color={"GrayText"}>Fetching requirements...</Text>
+    </VStack>
+  )
+
+  const { setValue } = useFormContext()
+
+  const handleSelect = (reqId: number) => {
+    setValue("dynamic.requirementId", reqId)
+    onSelect(reqId)
+  }
+
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={onClose} colorScheme={"dark"}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader pb={4}>Select base value</ModalHeader>
+          <ModalBody>
+            {dynamicRequirements?.length > 0 ? (
+              <>
+                <Text fontWeight={"semibold"} color={"GrayText"} mb={5}>
```

Comment:
It'd be great to give more context here that 1. we need requirements to provide values so they will influence the access to the role, and 2. only specific requirements provide values and are selectable